include/cudaq/Optimizer/CodeGen/Peephole.td

```diff
@@ -148,4 +148,33 @@ def MeasureToRegisterCallConv : Pat<
     (CreateMeasureCall $call, $args),
     [(IsaMeasureToRegisterCall:$callee), (IsaIntToPtrOperand $args)]>;
 
+//===----------------------------------------------------------------------===//
+def EraseMeasureOp : NativeCodeCall<"static_cast<std::size_t>(0)">;
```

Comment:
Maybe these types of operations would be easier to express as C++ `RewritePatterns` (if I'm reading this right you are replacing with just some null statement to get the TableGen to work).

Reply:
Yes, I am just replacing with some null statements, similar to what is done in lines 105-118 above. Is there a reason these should be done differently than those? I'm definitely ok with updating these ... just looking for a bit more info.

Reply:
Generally, I think we're moving away from using tablegen to generate rewrite patterns. It's a bit more code, but I think we agreed that it is easier to maintain as C++.

Reply:
This is what I had thought I remembered...

Reply:
OK, that makes sense. I updated the code (in [0893654](https://github.com/NVIDIA/cuda-quantum/pull/661/commits/08936549ab2f5342da263dca160681200b209e55)) to use `OpRewritePattern`s instead of the TableGen approach.